catalyst_voices/packages/catalyst_voices_services/lib/src/storage/vault/vault.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:catalyst_voices_services/src/storage/storage.dart';
+import 'package:catalyst_voices_services/src/storage/vault/lock_factor.dart';
+
+/// Secure version of [Storage] where any read/write methods can take
+/// effect only when [isUnlocked] returns true.
+///
+/// In order to unlock [Vault] sufficient [LockFactor] have to be
+/// set via [unlock] that can unlock [LockFactor] from [setLock].
+///
+/// See [LockFactor.unlocks] for more details.
+abstract interface class Vault implements Storage {
```

Comment:
StorageVault? Just Vault is quite generic

Reply:
I like `Vault`, seems okey for me. We may discuss it if you want

Reply:
that's fine, lets keep it